qualtran/bloqs/basic_gates/toffoli.py

```diff
@@ -127,6 +129,29 @@ def wire_symbol(self, reg: Optional[Register], idx: Tuple[int, ...] = tuple()) -
             return ModPlus()
         raise ValueError(f'Unknown wire symbol register name: {reg.name}')
 
+    def get_ctrl_system(self, ctrl_spec: 'CtrlSpec') -> Tuple['Bloq', 'AddControlledT']:
+        from qualtran.bloqs.basic_gates import CNOT
+        from qualtran.bloqs.mcmt import ControlledViaAnd
+
+        if ctrl_spec != CtrlSpec():
+            return super().get_ctrl_system(ctrl_spec)
+
+        cc_cnot = ControlledViaAnd(CNOT(), CtrlSpec(cvs=[1, 1]))
+
+        def add_controlled(
+            bb: 'BloqBuilder', ctrl_soqs: Sequence['SoquetT'], in_soqs: dict[str, 'SoquetT']
+        ) -> tuple[Iterable['SoquetT'], Iterable['SoquetT']]:
+            (new_ctrl,) = ctrl_soqs
+            ctrl0, ctrl1 = cast(NDArray, in_soqs.pop('ctrl'))
+
+            (new_ctrl, ctrl0), ctrl1, target = bb.add(
+                cc_cnot, ctrl2=np.array([new_ctrl, ctrl0]), ctrl=ctrl1, target=in_soqs.pop('target')
+            )
+
+            return [new_ctrl], [ctrl0, ctrl1, target]
+
+        return cc_cnot, add_controlled
+
```

Comment:
I was thinking the natural thing to do here should be to return an instance of the `MultiControlX` gate if you do `Toffoli().controlled()`,  but then I checked that we don't return `MultiControX` from `XGate.controlled()` or `CNOT.controlled()` either. 

This works for now. Eventually, it'll be nice to have the same bloq constructed upon calling 
```python
>>> XGate().controlled(CtrlSpec(cvs=[1, 1, 1, 1, 1]))
>>> CNOT().controlled(CtrlSpec(cvs=[1, 1, 1, 1]))
>>> Toffoli().controlled(CtrlSpec(cvs=[1, 1, 1]))
```

and the natural bloq to return for all three would be `MultiControlX(cvs=(1, 1, 1, 1, 1))`.

Reply:
fair enough -- I opened an issue to track